lib_json/src/main/x/json.x

```diff
@@ -25,4 +25,39 @@ module json.xtclang.org
      * JSON types include primitive types, array types, and map types.
      */
     typedef (Primitive | Map<String, Doc> | Array<Doc>) as Doc;
+
+    /**
+     * A type representing a JSON Object.
+     */
+    typedef Map<String, Doc> as JsonObject;
+
+    /**
+     * A type representing a JSON Array.
+     */
+    typedef Array<Doc> as JsonArray;
+
+    /**
+     * A type representing a non-primitive JSON structure.
+     */
+    typedef JsonObject | JsonArray as JsonStruct;
+
+    /**
+     * @return a new instance of a mutable `JsonObject`.
```

Comment:
Trivial thing - we usually don't put the dot ('.') at the end of @return or @param